Wangscape/logging/Logger.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "Level.h"
+
+#include <string>
+
+
+namespace logging
+{
+
+class Logger
+{
+public:
+    Logger() = default;
+    explicit Logger(std::ostream* out_)
+        : mAppender{out_}
+    {
+    }
+    void setStream(std::ostream* out)
+    {
+        mAppender = out;
+    }
+    template<typename T>
+    Logger& operator<<(T&& msg)
+    {
+        if (mAppender != nullptr) {
+            *mAppender << std::forward<T>(msg);
+        }
+        return *this;
+    }
+private:
+    std::ostream* mAppender = nullptr;
```

Comment:
How does this work when mAppender is a `std::ofstream*`? Who owns this pointer and is responsible for allocating and deallocating it? Is it possible to use a smart pointer without risking a `delete &std::cout` call?

It may be worth writing some tests to ensure that the logging system can't cause a crash.